api/src/main/java/com/graphhopper/util/PointList.java

```diff
@@ -399,7 +404,10 @@ public static PointList fromLineString(LineString lineString) {
     }
 
     public LineString toLineString(boolean includeElevation) {
-        GeometryFactory gf = new GeometryFactory();
+        return toLineString(new GeometryFactory(), includeElevation);
```

Comment:
Wouldn't it be better to create a factory for 2D and 3D each and keep them?

Reply:
Have done this (one static factory). Or why would we want 2?

Reply:
```java
GeometryFactory FAC_2D = new GeometryFactory(new PackedCoordinateSequenceFactory(DOUBLE, 2));
GeometryFactory FAC_3D = new GeometryFactory(new PackedCoordinateSequenceFactory(DOUBLE, 3));
```
`includeElevation` would just toggle which one of the two is being passed to the function. No need to create new objects here.

Reply:
Have adapted the current code (via a single factory).

Reply:
Or will there be something different with your proposed changed?

Reply:
Hmmh, maybe you mean something else? There is just one call and it is:
```
return factory.createLineString(new PackedCoordinateSequence.Double(coordinates, includeElevation ? 3 : 2));
```